src/firmware/app/primitives/test_util_world.h

```diff
@@ -0,0 +1,211 @@
+
+// Initializing all the mock functions required for robot and world
+// Mocks with no return types or arguments:
+// https://github.com/meekrosoft/fff#hello-fake-world
+// Mocks with input arguments:
+// https://github.com/meekrosoft/fff#capturing-arguments
+// Mocks with outputs:
+// https://github.com/meekrosoft/fff#return-values
+namespace FirmwareTestUtil
+{
+    // Mock fake charger functions
+    FAKE_VOID_FUNC(charge_capacitor);
+    FAKE_VOID_FUNC(discharge_capacitor);
+    FAKE_VOID_FUNC(float_capacitor);
+
+    // Mock fake chicker functions
+    FAKE_VOID_FUNC(set_kick_speed, float);
+    FAKE_VOID_FUNC(set_chip_distance, float);
+    FAKE_VOID_FUNC(enable_auto_kick, float);
+    FAKE_VOID_FUNC(enable_auto_chip, float);
+    FAKE_VOID_FUNC(disable_auto_chip);
+    FAKE_VOID_FUNC(disable_auto_kick);
+
+    // Mock fake dribbler functions
+    FAKE_VOID_FUNC(set_requested_rpm, uint32_t);
+    FAKE_VOID_FUNC(enable_coast);
+    FAKE_VALUE_FUNC(unsigned int, get_temperature_deg_c);
+
+    // Mock fake robot functions
+    FAKE_VALUE_FUNC(float, get_robot_property);
+
+    // Mock fake wheel functions
+    FAKE_VOID_FUNC(request_wheel_force_front_right, float);
+    FAKE_VALUE_FUNC(float, get_motor_speed_front_right);
+    FAKE_VOID_FUNC(brake_front_right);
+    FAKE_VOID_FUNC(coast_front_right);
+
+    FAKE_VOID_FUNC(request_wheel_force_front_left, float);
+    FAKE_VALUE_FUNC(float, get_motor_speed_front_left);
+    FAKE_VOID_FUNC(brake_front_left);
+    FAKE_VOID_FUNC(coast_front_left);
+
+    FAKE_VOID_FUNC(request_wheel_force_back_right, float);
+    FAKE_VALUE_FUNC(float, get_motor_speed_back_right);
+    FAKE_VOID_FUNC(brake_back_right);
+    FAKE_VOID_FUNC(coast_back_right);
+
+    FAKE_VOID_FUNC(request_wheel_force_back_left, float);
+    FAKE_VALUE_FUNC(float, get_motor_speed_back_left);
+    FAKE_VOID_FUNC(brake_back_left);
+    FAKE_VOID_FUNC(coast_back_left);
+
+    // Mock fake ball functions
+    FAKE_VALUE_FUNC(float, get_ball_property);
+};  // namespace FirmwareTestUtil
+
+// Mock wheel state
+WheelConstants_t wheel_constants = {.motor_current_per_unit_torque       = 1.1f,
+                                    .motor_phase_resistance              = 1.2f,
+                                    .motor_back_emf_per_rpm              = 1.3f,
+                                    .motor_max_voltage_before_wheel_slip = 1.4f,
+                                    .wheel_radius                        = 1.5f,
+                                    .wheel_rotations_per_motor_rotation  = 0.5f};
+
+// Mock controller state
+ControllerState_t controller_state = {.last_applied_acceleration_x       = 2.33f,
+                                      .last_applied_acceleration_y       = 1.22f,
+                                      .last_applied_acceleration_angular = 3.22f};
+
+// Mock robot constants
+RobotConstants_t robot_constants = {
+    .mass = 1.1f, .moment_of_inertia = 1.2f, .robot_radius = 1.3f, .jerk_limit = 1.4f};
```

Comment:
do we ever need to change these values? is it worth having a function for each of this in the fixture, a test can override if it needs to? (or someother way of "dependency injecting")

Reply:
ATM, I don't think we need to for any of the firmware primitive tests. In the future, it could be investigated.

Reply:
fair enough, we can figure out how to "fixturize" them when we get around to testing the controller if we need to